src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -231,6 +237,7 @@ parseLocalDeclaration :: TokenParser Declaration
 parseLocalDeclaration = positioned (P.choice
                    [ parseTypeDeclaration
                    , parseValueDeclaration
+                   , parseBoundValueDeclaration
```

Comment:
Well, `parseValueDeclaration` and `parseBoundValueDeclaration` overlap, so the parser has to backtrack and we can't guarantee good error messages. But we could make them share a parser, and process the result into the right sort of declaration. If we parse using `many1 parseBinderAtom`, then lists where the first element is a `VarBinder` represent `ValueDeclaration`s and singleton lists represent `BoundValueDeclaration`s. Everything else is an error, but now we don't need to backtrack.